addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -839,14 +867,135 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                }
+                // Favorite ID (in form xxxxxxxxx.n)
+                else if (part.startsWith("id%3A")) {
+                    id = part.substring("id%3A".length());
+                    f = new Favorite(id);
+                    favorites.add(f);
+                }
+                // Favorite name
+                else if (part.startsWith("name%3A")) {
+                    name = decode(part.substring("name%3A".length()));
+                    if (f != null) {
+                        f.name = name;
+                    }
+                }
+                // Favorite type
+                else if (part.startsWith("type%3A")) {
+                    type = decode(part.substring("type%3A".length()));
+                    if (f != null) {
+                        f.type = type;
+                    }
+                }
+                // Favorite is audio
+                else if (part.startsWith("isaudio%3A")) {
+                    isaudio = new Boolean("1".matches(part.substring("isaudio%3A".length())));
+                    if (f != null) {
+                        f.isaudio = isaudio;
+                    }
+                }
+                // When "1", favorite is a submenu with additional favorites
+                else if (part.startsWith("hasitems%3A")) {
+                    hasitems = new Boolean("1".matches(part.substring("hasitems%3A".length())));
+                    if (f != null) {
+                        if (hasitems) {
+                            // Skip subfolders
+                            favorites.remove(f);
+                            f = null;
+                        } else {
+                            f.hasitems = hasitems;
+                        }
+                    }
+                }
+            }
+            updatePlayersFavoritesList(favorites);
+            updateChannelFavoritesList(favorites);
+        }
+
+        private void updatePlayersFavoritesList(List<Favorite> favorites) {
+            updatePlayer(new PlayerUpdateEvent() {
+                @Override
+                public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                    listener.updateFavoritesListEvent(favorites);
+                }
+            });
+        }
+
+        private void updateChannelFavoritesList(List<Favorite> favorites) {
+            // Get config parameter indicating whether name should be wrapped with double quotes
+            final boolean includeQuotes = getConfigAs(SqueezeBoxServerConfig.class).quoteFavoritesList;
+            final String quote = includeQuotes ? "\"" : "";
+            String adjustedName;
+
+            StringBuilder sb = new StringBuilder();
+            for (Favorite favorite : favorites) {
+                // If not quoting, we don't want any embedded commas
+                adjustedName = includeQuotes ? favorite.name : favorite.name.replaceAll(",", "");
+                sb.append(favorite.shortId + "=");
```

Comment:
Avoid concatenation of strings, if you use a `StringBuilder`. Use the `append` method all the time.